### PR TITLE
chore: front-end-avatars

### PR DESCRIPTION
**Pull request by sean-brydon**
Fixes front-end avatars being a bit all over the place. 

Logically made no sense to use OrganizationAvatar component for every single UserAvatar we require the org logo. 

Refactored to include this in UserAvatar which i think makes more sense

App remains unchanged - no UI/UX changes have been implemented in this PR 

CleanShot 2023-12-10 at 06 45 50@2x

CleanShot 2023-12-10 at 06 45 54@2x

### CODE REVIEW

File: packages/ui/components/avatar/UserAvatar.test.tsx
Comment (sean-brydon) [inline packages/ui/components/avatar/UserAvatar.test.tsx:1]: Benefit of dumb components we can test to make sure they work

File: packages/ui/components/avatar/UserAvatar.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ * It is aware of the user's organization to correctly show the avatar from the correct URL
+ */
+export function UserAvatar(props: UserAvatarProps) {
+  const { user, previewSrc = getUserAvatarUrl(user), ...rest } = props;
+
+  const indicator = props.organization ? (
+    <OrganizationIndicator size={props.size} organization={props.organization} user={props.user} />
+  ) : (
+    props.indicator
+  );
```
Comment (sean-brydon) [inline packages/ui/components/avatar/UserAvatar.tsx:50]: If we pass in an organisation we use that to calculate the indicator

**Comment by Udit-takkar** [comment]
will this also fix this ?

https://github.com/calcom/cal.com/assets/53316345/b4b1204e-1350-4201-b68c-8defc0f67347

**Comment by PeerRich** [comment]
this looks like a good idea

@sean-brydon whats missing here to undraft it?

File: packages/ui/components/avatar/UserAvatar.test.tsx [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+import { UserAvatar } from "./UserAvatar";
+
+const mockUser = {
+  name: "John Doe",
+  username: "pro",
+  organizationId: null,
+};
+
+describe("tests for UserAvatar component", () => {
```
Comment (sean-brydon) [inline packages/ui/components/avatar/UserAvatar.test.tsx:14]: Unit tests for some react components 👀

File: packages/ui/components/avatar/UserAvatar.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  organization?: Organization | null;
+  alt?: string | null;
+};
+
+function OrganizationIndicator({
+  size,
+  organization,
+  user,
+}: Pick<UserAvatarProps, "size" | "user"> & { organization: Organization }) {
+  const organizationUrl = organization.logoUrl ?? getOrgAvatarUrl(organization);
```
Comment (emrysal) [inline packages/ui/components/avatar/UserAvatar.tsx:28]: Nice 😎

**Review by emrysal — APPROVED** [review]
Looks good to me, will merge tomorrow unless someone beats me to it.

**Pull request merged**